src/ui/alert/Alert.tsx

```diff
@@ -1,50 +1,38 @@
-import React, { useCallback, useState, memo } from 'react';
-
-import Snackbar from '@material-ui/core/Snackbar';
-import CloseIcon from '@material-ui/icons/Close';
-
-import { Button } from '..';
+import React, { useState, useEffect } from 'react';
 
 import csx from './Alert.scss';
 
 namespace Alert {
-  export type Types = 'warning' | 'error' | 'success' | 'info';
-
   export interface Props {
-    message: string;
-    type?: Types;
-    onClose(): void;
+    message?: string;
+    type?: 'warning' | 'error' | 'success' | 'info';
+    className?: string;
+    time?: number;
+  }
+}
+
+const Alert = ({ message, type = 'error', className = "", time = 5000 }: Alert.Props) => {
```

Comment:
This should be presentational component only - all logic to - show / hide alerts should be in Provider.

How to implement:

- if error occurs show - alert set delay 
- If next error occurs - show after first one disappear.
- use RxJs operators  for that concatMap, delay, takeUntil and others

Reply:
All alert logic was replaced to AlertProvider, now it works on a recursive basis: function checks if any message is currently displayed, if so, it waits 300ms (cause it is animation time) and checks again, if not, it displays.